spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTests.java

```diff
@@ -0,0 +1,229 @@
+		poolingOptionsFactoryBean.setIdleTimeoutSeconds(120);
+		poolingOptionsFactoryBean.setInitializationExecutor(mockExecutor);
+		poolingOptionsFactoryBean.setLocalCoreConnections(50);
+		poolingOptionsFactoryBean.setLocalMaxConnections(1000);
+		poolingOptionsFactoryBean.setLocalMaxSimultaneousRequests(200);
+		poolingOptionsFactoryBean.setLocalMinSimultaneousRequests(100);
+		poolingOptionsFactoryBean.setPoolTimeoutMilliseconds(300);
+		poolingOptionsFactoryBean.setRemoteCoreConnections(25);
+		poolingOptionsFactoryBean.setRemoteMaxConnections(250);
+		poolingOptionsFactoryBean.setRemoteMaxSimultaneousRequests(100);
+		poolingOptionsFactoryBean.setRemoteMinSimultaneousRequests(50);
+
+		assertThat(poolingOptionsFactoryBean.getHeartbeatIntervalSeconds(), is(equalTo(15)));
+		assertThat(poolingOptionsFactoryBean.getIdleTimeoutSeconds(), is(equalTo(120)));
+		assertThat(poolingOptionsFactoryBean.getInitializationExecutor(), is(equalTo(mockExecutor)));
+		assertThat(poolingOptionsFactoryBean.getLocalCoreConnections(), is(equalTo(50)));
+		assertThat(poolingOptionsFactoryBean.getLocalMaxConnections(), is(equalTo(1000)));
+		assertThat(poolingOptionsFactoryBean.getLocalMaxSimultaneousRequests(), is(equalTo(200)));
+		assertThat(poolingOptionsFactoryBean.getLocalMinSimultaneousRequests(), is(equalTo(100)));
+		assertThat(poolingOptionsFactoryBean.getPoolTimeoutMilliseconds(), is(equalTo(300)));
+		assertThat(poolingOptionsFactoryBean.getRemoteCoreConnections(), is(equalTo(25)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMaxConnections(), is(equalTo(250)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMaxSimultaneousRequests(), is(equalTo(100)));
+		assertThat(poolingOptionsFactoryBean.getRemoteMinSimultaneousRequests(), is(equalTo(50)));
+	}
+
+	@Test
+	public void afterPropertiesSetInitializesLocalPoolingOptions() throws Exception {
+		PoolingOptionsFactoryBean poolingOptionsFactoryBean = new PoolingOptionsFactoryBean() {
+			@Override PoolingOptions newPoolingOptions() {
+				return mockPoolingOptions;
+			}
+		};
+
+		when(mockPoolingOptions.getNewConnectionThreshold(eq(HostDistance.LOCAL))).thenReturn(1);
+
+		poolingOptionsFactoryBean.setHeartbeatIntervalSeconds(60);
+		poolingOptionsFactoryBean.setIdleTimeoutSeconds(300);
+		poolingOptionsFactoryBean.setLocalCoreConnections(10);
+		poolingOptionsFactoryBean.setLocalMaxConnections(100);
+		poolingOptionsFactoryBean.setLocalMaxSimultaneousRequests(50);
+		poolingOptionsFactoryBean.setLocalMinSimultaneousRequests(5);
+		poolingOptionsFactoryBean.setPoolTimeoutMilliseconds(180);
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(nullValue(PoolingOptions.class)));
+
+		poolingOptionsFactoryBean.afterPropertiesSet();
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(sameInstance(mockPoolingOptions)));
+		assertThat(poolingOptionsFactoryBean.getObjectType(), is(equalTo((Class) mockPoolingOptions.getClass())));
+
+		InOrder inOrder = inOrder(mockPoolingOptions);
```

Comment:
Yeah, I contemplated this a lot.

Personally, I feel it is overkill to express this in XML and actually start an actual Cassandra instance (embedded or not), just to verify the DataStax `PoolingOptions` instance.  Using actual XML is good for actual test case(s) involving and testing the parsers, but even then I question whether an actual instance of (embedded?) Cassandra is needed for parser tests either.

SDG makes use of mocking, even at the integration test level, with the use of a _Spring_ [ApplicationContextInitializer](https://github.com/jxblum/spring-data-gemfire/blob/master/src/test/java/org/springframework/data/gemfire/test/GemfireTestApplicationContextInitializer.java) and the help of the [GemfireTestBeanPostProcessor](https://github.com/jxblum/spring-data-gemfire/blob/master/src/test/java/org/springframework/data/gemfire/test/GemfireTestBeanPostProcessor.java) which creates mock versions of the SDG `FactoryBeans` to create mock GemFire components (e.g. Cache) instead of actual GemFire components (i.e. `GemFireCacheImpl`).

This is especially nice for testing the XML namespace elements where it is only important that the configuration meta-data used to configure and initialize the underlying data store components actually occurred and was proper (according to the API), not whether the actual underlying data store components behave correctly according to the configuration.

All of this is to say that I would eventually like to employ similar testing patterns and techniques to SD Cassandra to keep the test suite as light and fast as possible, thus relying more on "unit" tests over "integration" tests when the functionality/behavior can adequately be covered in unit tests where possible.

In this particular case, that is exactly what the [existing test case method](https://github.com/spring-projects/spring-data-cassandra/blob/master/spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTest.java#L45-L68) was testing, an actual instance of the `PoolingOptions` class in order to address bug [DATACASS-176](https://jira.spring.io/browse/DATACASS-176) (see also [here](https://github.com/spring-projects/spring-data-cassandra/blob/master/spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTest.java#L43) along with the [comment](https://github.com/spring-projects/spring-data-cassandra/blob/master/spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTest.java#L39-L40)).

I had half a thought of rewriting this test case method since...

1. What is [this](https://github.com/spring-projects/spring-data-cassandra/blob/master/spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTest.java#L59-L66) really testing? (argh)

2. I mean, if you are going to call [afterPropertiesSet()](https://github.com/spring-projects/spring-data-cassandra/blob/master/spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTest.java#L57) without mocks, you might as well test the actual `PoolingOptions` instance created, which is exactly how I modified this test.

I thought about adding order verification here, but I decided not.

In addition, I have cleaned up all instances of my order verification use in my additional test cases in the updated PR.

Let me know if you have any additional thoughts on this.  Thx.